### PR TITLE
tests: sensor generic: update to new ZTEST APIs

### PR DESCRIPTION
**Pull request by aaronemassey**
Updates tests to use the new ZTEST APIs

Signed-off-by: Aaron Massey <aaronmassey@google.com>

Part of https://github.com/zephyrproject-rtos/zephyr/issues/47002 effort

### CODE REVIEW

File: tests/drivers/sensor/generic/src/main.c [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 #include <ztest.h>
 #include "dummy_sensor.h"
+#include "toolchain/gcc.h"
```
Comment (teburd) [inline tests/drivers/sensor/generic/src/main.c:16]: This and the next include shouldn't be needed?

**Review by teburd — CHANGES_REQUESTED** [review]

File: tests/drivers/sensor/generic/src/main.c (hunk 1 above)
Comment (aaronemassey) [reply tests/drivers/sensor/generic/src/main.c]: Whoops. ClangD auto-inserted and missed it. Thanks!

**Review by teburd — DISMISSED** [review]

**Review by gmarull — DISMISSED** [review]

File: tests/drivers/sensor/generic/src/main.c [hunk 2]
```diff
@@ -103,7 +103,7 @@ static struct trigger_sequence trigger_elements[] = {
  *
  * @see sensor_sample_fetch(), sensor_channel_get()
  */
-void test_sensor_get_channels(void)
+ZTEST(test_sensor_api, test_sensor_get_channels)
```
Comment (nashif) [inline tests/drivers/sensor/generic/src/main.c:106]: lets drop the leading test_ from the suite name.

Comment (aaronemassey) [reply tests/drivers/sensor/generic/src/main.c]: Done.

**Review by teburd — DISMISSED** [review]

**Review by MaureenHelm — DISMISSED** [review]

**Review by chen-png — DISMISSED** [review]

**Review by nashif — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**